week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,71 @@
-function setAlarm() {}
+function setAlarm() {
+  //get the value of the input field
+  let inputField = document.querySelector("#alarmSet").value;
+
+  //set the title to the correct value
+  const timeRemaining = document.querySelector("#timeRemaining");
+  timeRemaining.textContent = `Time Remaining: ${inputField}:00`;
+
+  let myAlarm;
+  ///
+  document.getElementById("set").hidden = true;
+
+  // add `pause` functionality
+  document.getElementById("set").textContent = "Pause";
+
+  //create new button Pause
+  let pauseButton = document.createElement("button");
+  pauseButton.textContent = "Pause";
+  let divButton = document.querySelector(".centre >:last-child");
+  divButton.prepend(pauseButton);
+  pauseButton.className = "pauseButton";
+
+  document.querySelector(".pauseButton").addEventListener("click", () => {
+    clearInterval(myAlarm);
+    document.querySelector(".pauseButton").hidden = true;
+
+    let restartButton = document.createElement("button");
+    restartButton.textContent = "Restart";
+    let divButton = document.querySelector(".centre >:last-child");
+    divButton.prepend(restartButton);
+    restartButton.className = "restartButton";
+
+    //restart functionality
+    document.querySelector(".restartButton").addEventListener("click", () => {
+      location.reload();
+    });
+  });
+
+  //update the `Time Remaining` title every second
+  myAlarm = setInterval(() => {
+    if (inputField == 0) {
+      clearInterval(myAlarm);
+      //play the alarm sound
+      playAlarm();
+
+      //Make the background change color when the alarm clock finishes
+      setTimeout(() => {
+        document.body.style.backgroundColor = "orange";
+        //making the background flash
+        let index = 0;
+        setInterval(function () {
+          if (index % 2) {
+            document.body.style.backgroundColor = "white";
+            index++;
+          } else {
+            document.body.style.backgroundColor = "orange";
+            index++;
+            timeRemaining.textContent = `Time Remaining: ${inputField}:00`;
+          }
+        }, 1000);
```

Comment:
This is good. If you look up the modulus operator (`%`) you might get an idea of how you could have an array of colours of any length you want and iterate through them on each interval with somewhat less code. This is strictly an extension for if you have time though, rather than something to get too involved in.

See here for an introduction to the modulus: https://riptutorial.com/javascript/example/760/remainder---modulus----

Reply:
Thanks for your feedback. I've read the article about modulus operator, but I've not understood yet. 
But I can form the array of colours just calling function like this:
`function randomColor() {
  let letters = "abcdef1234567890";
  let color = "#";
  for (let i = 0; i < 6; i++) {
    color = color + letters[Math.floor(Math.random() * 16)];
  }
  return color;
}`
But it is interesting question, I will think about it in the background.
Thanks for your assistance.